src/common/PropertyExt.js

```diff
@@ -154,16 +154,36 @@
     };
 
     // Publish Properties  ---
-    PropertyExt.prototype.publishedProperties = function (includePrivate) {
+    PropertyExt.prototype.publishedProperties = function (includePrivate, expandProxies) {
         var retVal = [];
         for (var key in this) {
             if (isMeta(key) && (includePrivate || !isPrivate(this, key))) {
-                retVal.push(this[key]);
+                var meta = this[key];
+                if (expandProxies && meta.type) {
+                    var item = this;
+                    while (meta.type === "proxy") {
+                        item = item[meta.proxy];
+                        meta = item.publishedProperty(meta.method);
+                    }
+                    if (meta.id !== this[key].id) {
+                        meta = JSON.parse(JSON.stringify(meta));  //  Clone meta so we can safely replace the id.
```

Comment:
@jbrundage:
1.  This is the same code that is currently in use, just refactored from Persist.js to PropertyExt.js
2. We don't don't serialize function properties that way (and our Widget Arrays shouldn't have circular references).

There probably is a more efficient way to do this, but I would rather leave the existing behavior in place and if we do change it, do it in a diff pull request?